### DISCUSSION TITLE
Partial enums for source generators

### DISCUSSION THREAD

**Discussion by tuscen** [dotnet/csharplang#6090] [Language Ideas]
I have a project where we generate custom json converters for enums using source generators. Right now we have to mark serializable enums using generated converters manually which leads to errors during source editing before first compilation.

It would be really convenient if enums could be declared partial so we could generate code like that.

Original source code:

```csharp
public partial enum Foo
{
    Bar,
    Baz,
}
```

Generated source code:

```csharp
internal class FooEnumConverter : JsonConverter<Foo> 
{
    // ...
}

[JsonConverter(typeof(FooEnumConverter))]
public partial enum Foo
{}
```

Partial enums can work in such a way that type member declarations should be allowed only in a single class declaration, but type annotations will be allowed everywhere.

**Comment by HaloFour** [comment]
In this case only one of the parts could contain the actual members of the enum?

**Reply by tuscen** [reply]
Yes, that way there won't be a problem with members ordering ambiguity
